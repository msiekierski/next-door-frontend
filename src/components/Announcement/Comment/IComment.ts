interface IComment {
  idAccount?: number;
  idAnnouncement: number;
  creationDate: string;
  description: string;
}

export default IComment;
