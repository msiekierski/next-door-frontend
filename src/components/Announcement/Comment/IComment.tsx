interface IComment {
  id: number;
  author: string;
  date: number;
  description: string;
}

export default IComment;
