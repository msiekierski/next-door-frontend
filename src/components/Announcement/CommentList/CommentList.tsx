import React, { FunctionComponent } from "react";
import { ListGroup } from "react-bootstrap";
import IComment from "../Comment/IComment";
import Comment from "../Comment/Comment";

interface OwnProps {
  comments: Array<IComment>;
}

type Props = OwnProps;

const CommentList: FunctionComponent<Props> = ({ comments }) => {
  return (
    <ListGroup>
      {comments
        .sort((a, b) => Number(new Date(b.creationDate)) - Number(new Date(a.creationDate)))
        .map((comment, index) => (
          <Comment
            key={index}
            idAccount={comment.idAccount}
            idAnnouncement={comment.idAnnouncement}
            creationDate={comment.creationDate}
            description={comment.description}
          />
        ))}
    </ListGroup>
  );
};

export default CommentList;
