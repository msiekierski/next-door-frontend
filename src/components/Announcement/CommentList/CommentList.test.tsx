import React from "react";
import renderer from "react-test-renderer";
import IComment from "../Comment/IComment";
import CommentList from "./CommentList";

const comments: Array<IComment> = [
  {
    creationDate: new Date().toDateString(),
    description: "Hej",
    idAccount: 0,
    idAnnouncement: 0,
  },
  {
    creationDate: new Date().toDateString(),
    description: "Hello",
    idAccount: 0,
    idAnnouncement: 0,
  },
  {
    creationDate: new Date().toDateString(),
    description: "Welcome!",
    idAccount: 0,
    idAnnouncement: 0,
  },
];

it("renders correctly empty", () => {
  const tree = renderer.create(<CommentList comments={[]} />);
  expect(tree).toMatchSnapshot();
});

it("renders correctly with comments", () => {
  const tree = renderer.create(<CommentList comments={comments} />);
  expect(tree).toMatchSnapshot();
});
