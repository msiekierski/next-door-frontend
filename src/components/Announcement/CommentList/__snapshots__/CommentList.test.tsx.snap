// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
/>
`;

exports[`renders correctly with comments 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="d-flex justify-content-between card-title h5"
      >
        <div
          className="text-muted card-subtitle h6"
        >
          Author
        </div>
        <div
          className="text-muted text-right card-subtitle h6"
        >
          2021-1-22
          <br />
          0:00:00
        </div>
      </div>
      <p
        className="card-text"
      >
        Hej
      </p>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="d-flex justify-content-between card-title h5"
      >
        <div
          className="text-muted card-subtitle h6"
        >
          Author
        </div>
        <div
          className="text-muted text-right card-subtitle h6"
        >
          2021-1-22
          <br />
          0:00:00
        </div>
      </div>
      <p
        className="card-text"
      >
        Hello
      </p>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="d-flex justify-content-between card-title h5"
      >
        <div
          className="text-muted card-subtitle h6"
        >
          Author
        </div>
        <div
          className="text-muted text-right card-subtitle h6"
        >
          2021-1-22
          <br />
          0:00:00
        </div>
      </div>
      <p
        className="card-text"
      >
        Welcome!
      </p>
    </div>
  </div>
</div>
`;
